src/main/java/bio/terra/service/dataset/flight/ingest/IngestUtils.java

```diff
@@ -133,4 +137,20 @@ public static PdaoLoadStatistics getIngestStatistics(FlightContext context) {
     FlightMap workingMap = context.getWorkingMap();
     return workingMap.get(IngestMapKeys.INGEST_STATISTICS, PdaoLoadStatistics.class);
   }
+
+  public static Predicate<FlightContext> noFilesToIngestPredicate() {
+    return flightContext -> {
+      if (flightContext.getFlightClassName().equals(DatasetIngestFlight.class.getName())
+          && Optional.ofNullable(
+                  flightContext.getWorkingMap().get(IngestMapKeys.BULK_LOAD_FILE_MODELS, Set.class))
+              .map(Set::isEmpty)
+              .orElse(true)) {
+        Logger logger = LoggerFactory.getLogger(DatasetIngestFlight.class);
+        logger.info(
+            "Skipping {} because there are no files to ingest", flightContext.getStepClassName());
+        return true;
+      }
+      return false;
```

Comment:
This result confuses me; in the case where it's not a `DatasetIngestFlight`, are there always files to ingest?

Reply:
Not all the time. If we get a request that has files in it that have all already been ingested, or if this is a dataset ingest without files, then there are no files to actually ingest.